diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -606,10 +618,11 @@ namespace diff_drive_controller
       }
     }
 
+    // Compute the period estimated from time:
+    const ros::Duration period_estimated = time - time_previous_;
+
     // Compute/Set control period and frequency (desired/expected or real):
-    // @todo also allow to compute the period as time - time_previous_, because
-    // they can be different
-    const double control_period    = control_period_desired_    > 0.0 ? control_period_desired_    : period.toSec();
+    const double control_period    = control_period_desired_    > 0.0 ? control_period_desired_    : (period_from_time_ ? period_estimated.toSec() : period.toSec());
     const double control_frequency = control_frequency_desired_ > 0.0 ? control_frequency_desired_ : 1.0 / control_period;
```

Comment:
That's fair. It would be great if realtime_tools also had a set of ROS_RT_XX macros for dealing with logging, though I guess the real issue is that even just storing strings to get them onto the other thread incurs memory allocations. Maybe they could be non-formatting versions, so you're really just tossing const char* pointers on to a ring buffer or something?

In this immediate instance, it could be as simple as set flag, output message, clear flag. But as you say, if proper diagnostics are coming soon enough, then whatever.